src/mono/mono/eventpipe/ep-rt-mono.h

```diff
@@ -844,8 +850,7 @@ ep_rt_walk_managed_stack_for_thread (
 	ep_rt_thread_handle_t thread,
 	EventPipeStackContents *stack_contents)
 {
-	// TODO: Implement.
-	return true;
+	return ep_rt_mono_func_table_get ()->ep_rt_mono_walk_managed_stack_for_thread (thread, stack_contents) ? true : false;
```

Comment:
nit:
1. `expr ? true : false` can just be `expr` 
2. should be `TRUE` and `FALSE` - we generally don't use C99 bools (for some reason).

here and in other places

Reply:
EventPipe library uses C99 types so return of ep_rt_walk_managed_stack_for_thread should be true/false. Did an explicit conversion like this since called method (ep_rt_mono_walk_managed_stack_for_thread) implemented in mono runtime uses gboolean and wanted to explicitly convert and believe we got some compile errors/warnings in the past for implicit conversion between gboolean and bool. An option could be to do cast it to bool instead of expr ? true : false.

Reply:
ah ok. I checked the return type of the mono func, but didn't look at the EP function. This makes sense then. Unfortunate, but I guess your way is clearest.